classes/object_file_system.php

```diff
@@ -661,4 +661,23 @@ protected function copy_from_external_to_local($contenthash) {
         return copy($externalpath, $localpath);
     }
 
+    /**
+     * Generates signed URL to external file from its hash.
+     *
+     * @param string $contenthash File content hash.
+     *
+     * @return string.
+     */
+
+    public function generate_signed_url_to_external_file_from_hash($contenthash) {
+
+        if ($this->externalclient->support_signed_urls()) {
+            $signedurl = $this->externalclient->generate_signed_url($contenthash);
+        } else {
+            $signedurl = 'Pre-Signed URLs not supported for current storage';
```

Comment:
I think remove this and return null, we never want that string accidentally leaking in a url rewrite